cmd/broker/ingress/main.go

```diff
@@ -28,28 +28,41 @@ import (
 	"sync"
 	"time"
 
+	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+
 	cloudevents "github.com/cloudevents/sdk-go"
+	cehttp "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/http"
 	"github.com/kelseyhightower/envconfig"
 	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
 	"github.com/knative/eventing/pkg/broker"
 	"github.com/knative/eventing/pkg/provisioners"
+	"github.com/knative/eventing/pkg/tracing"
 	"github.com/knative/eventing/pkg/utils"
+	"github.com/knative/pkg/configmap"
 	"github.com/knative/pkg/signals"
+	pkgtracing "github.com/knative/pkg/tracing"
 	"go.opencensus.io/exporter/prometheus"
 	"go.opencensus.io/stats"
 	"go.opencensus.io/stats/view"
 	"go.opencensus.io/tag"
 	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	crlog "sigs.k8s.io/controller-runtime/pkg/runtime/log"
-	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
-	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	NAMESPACE = "NAMESPACE"
 )
 
 type envConfig struct {
-	Broker  string `envconfig:"BROKER" required:"true"`
-	Channel string `envconfig:"CHANNEL" required:"true"`
+	Broker    string `envconfig:"BROKER" required:"true"`
+	Channel   string `envconfig:"CHANNEL" required:"true"`
+	Namespace string `envconfig:"NAMESPACE" required:"true"`
+
+	ZipkinServiceName string `envconfig:"ZIPKIN_SERVICE_NAME" required:"true"`
```

Comment:
Why is this a required env var?

Reply:
It is the name that this binary will use in Zipkin traces. It is required, because I want it to be present and have ensured the Broker controller will provide it.

Added a comment in the code to say what it represents.

Reply:
The ingress binary could calculate this on its own since it also has broker name and namespace as required env vars. What's the advantage of putting this in an env var provided by the controller versus writing a function that generates it in process? It seems less complex to generate it here.